cmake/CMakeLists.txt

```diff
@@ -225,6 +225,13 @@ option(onnxruntime_ENABLE_EXTERNAL_CUSTOM_OP_SCHEMAS "Enable registering user de
 
 set(ONNX_CUSTOM_PROTOC_EXECUTABLE "" CACHE STRING "Specify custom protoc executable to build ONNX")
 
+# Enable retries on connection time out and TLS handshake failures for older CMake versions
+# This is implemented in latter versions of CMake, but is a nice-to-have when building
+# with older versions, since onnxruntime pulls in lots of dependencies!
+# 35 is in the default download_retry_codes from v3.28.5 onwards.
+# 28 is in the default download_retry_codes from v3.26.1 onwards
```

Comment:
we already require at least CMake 3.28.

https://github.com/microsoft/onnxruntime/blob/413d7b403c9e774acdbd204999bd93e0c2650261/cmake/CMakeLists.txt#L6

if you need to use an older CMake version (and override that minimum CMake version), can you specify this with `cmake ... -Ddownload_retry_codes="7;6;8;15;28;35"` (or the `build.py --cmake_extra_defines` option)?